packages/headless/src/controllers/core/facets/range-facet/headless-core-range-facet.test.ts

```diff
@@ -146,4 +145,146 @@ describe('range facet', () => {
+        expect(rangeFacet.state.values).toEqual([]);
+      });
+    });
+
+    describe('.sortCriterion', () => {
+      it('should expose the sortCriteria from the request', () => {
+        const criterion = 'descending';
+        props.getRequest = () =>
+          buildMockNumericFacetRequest({sortCriteria: criterion});
+        initRangeFacet();
+
+        expect(rangeFacet.state.sortCriterion).toBe(criterion);
+      });
+    });
+
+    describe('.resultsMustMatch', () => {
+      it('should expose the resultsMustMatch value from the request', () => {
+        const resultsMustMatch = 'allValues';
+        props.getRequest = () =>
+          buildMockNumericFacetRequest({resultsMustMatch});
+        initRangeFacet();
+
+        expect(rangeFacet.state.resultsMustMatch).toBe(resultsMustMatch);
+      });
+    });
+
+    describe('.hasActiveValues', () => {
+      it('should be true when #state.values has a value with a non-idle state', () => {
+        const values = [buildMockNumericFacetValue({state: 'selected'})];
+        const response = buildMockNumericFacetResponse({facetId, values});
+        state.search.response.facets = [response];
+
+        expect(rangeFacet.state.hasActiveValues).toBe(true);
+
+        values.pop();
+        values.push(buildMockNumericFacetValue({state: 'excluded'}));
+        state.search.response.facets = [
+          buildMockNumericFacetResponse({facetId, values}),
+        ];
+
+        expect(rangeFacet.state.hasActiveValues).toBe(true);
+      });
+
+      it('should be false when #state.values only has idle values', () => {
+        const values = [buildMockNumericFacetValue({state: 'idle'})];
+        const response = buildMockNumericFacetResponse({facetId, values});
+        state.search.response.facets = [response];
+
+        expect(rangeFacet.state.hasActiveValues).toBe(false);
+      });
+
+      it('should be false when #state.values is an empty array', () => {
+        expect(rangeFacet.state.hasActiveValues).toBe(false);
+      });
+    });
+
+    describe('.isLoading', () => {
+      it('should be true when search is loading', () => {
+        state.search.isLoading = true;
+        initRangeFacet();
+
+        expect(rangeFacet.state.isLoading).toBe(true);
+      });
+
+      it('should be false when search is not loading', () => {
+        state.search.isLoading = false;
+        initRangeFacet();
+
+        expect(rangeFacet.state.isLoading).toBe(false);
+      });
+    });
+
+    describe('.enabled', () => {
+      it('should be true when the facet is enabled', () => {
+        state.facetOptions.facets[facetId] = {enabled: true};
+        initRangeFacet();
+
+        expect(rangeFacet.state.enabled).toBe(true);
+      });
+
+      it('should be false when the facet is disabled', () => {
+        state.facetOptions.facets[facetId] = {enabled: false};
+        initRangeFacet();
+
+        expect(rangeFacet.state.enabled).toBe(false);
+      });
+    });
+
+    describe('.domain', () => {
```

Comment:
These are the tests for the feature added in this PR.

The remaining changes in this file are boy scouting: reorganizing tests in a more consistent fashion, and adding missing tests to complete the coverage of the controller.